meta-lmp-base/classes/lmp-staging.bbclass

```diff
@@ -12,6 +12,13 @@ LMPSTAGING_LOCK_TO_AVOID_OOM = "clang-native rust-native rust-llvm-native"
 python __anonymous() {
     pn = d.getVar('PN')
 
+    d.setVar('LMPSTAGING_KERN_ADD_ST_SUBDIR',  '')
```

Comment:
move this variable outside the function with the default version make the things more clear in my opinion

Reply:
Do you mean to move outside the function but leave it in the stage class?

Reply:
yes, I mean together with the LMPSTAGING_INHERIT_KERNEL_MODSIGN and LMPSTAGING_LOCK_TO_AVOID_OOM.

Reply:
I think this was a generic change done by the kernel for all arm32 targets, right? If so we can have the variable defined as arm32 instead of ST.

Reply:
> I think this was a generic change done by the kernel for all arm32 targets, right? If so we can have the variable defined as arm32 instead of ST.

For now, we need to handle only an ST vendor case, as NXP boards are handled in meta-freescale. As a prefix is vendor-specific, I'd keep it as is. Anytime we can rename it to a generic one if we have to expand a solution for other vendors.

Reply:
> yes, I mean together with the LMPSTAGING_INHERIT_KERNEL_MODSIGN and LMPSTAGING_LOCK_TO_AVOID_OOM.

Done, thanks!

Reply:
> > I think this was a generic change done by the kernel for all arm32 targets, right? If so we can have the variable defined as arm32 instead of ST.
> 
> For now, we need to handle only an ST vendor case, as NXP boards are handled in meta-freescale. As a prefix is vendor-specific, I'd keep it as is. Anytime we can rename it to a generic one if we have to expand a solution for other vendors.

True, we need to know the prefix.